src/lib/components/MainPanel.js

```diff
@@ -66,18 +61,12 @@ export default class MainPanel extends React.Component {
     scheduleSnoozedTab(time);
   }
 
-  handleTimeSelect(value) {
-    this.setState({ dateChoice: value });
-  }
-
   closeTimeSelect() {
     this.setState({ datepickerActive: false });
   }
 
-  confirmTimeSelect() {
-    const { dateChoice } = this.state;
+  confirmTimeSelect(dateChoice) {
     const { scheduleSnoozedTab } = this.props;
-
     if (!dateChoice) { return; }
```

Comment:
Just did a quick hacky update to try to ensure the wake timer is no earlier than 5 seconds from now.